src/util/imagefiledata.cpp

```diff
@@ -6,20 +6,20 @@
 #include <QFileInfo>
 #include <QImage>
 
-ImageFileData::ImageFileData(const QString& coverArtFilePath)
-        : QImage(coverArtFilePath) {
-    m_coverArtExtension = QFileInfo(coverArtFilePath).suffix();
-    QFile coverArtFile = QFile(coverArtFilePath);
-    coverArtFile.open(QIODevice::ReadOnly);
-    m_coverArtBytes = coverArtFile.readAll();
-    coverArtFile.close();
-}
-
 // Most of the cover art extensions are 'jpg' at cover art archive.
 ImageFileData::ImageFileData(const QByteArray& coverArtBytes)
         : QImage(QImage::fromData(coverArtBytes)), m_coverArtBytes(coverArtBytes) {
 }
 
+//static
+ImageFileData ImageFileData::fromFilePath(const QString& coverArtFilePath) {
+    QFile coverArtFile = QFile(coverArtFilePath);
+    coverArtFile.open(QIODevice::ReadOnly);
+    QByteArray coverArtBytes = coverArtFile.readAll();
+    coverArtFile.close();
+    return ImageFileData(coverArtBytes);
```

Comment:
The part of detecting the file extension is missing here. 

My original idea was to use something like: 

```
    QBuffer b = coverArtBytes;
    b.open(QIODevice::ReadOnly);
    auto reader = QImageReader(&b);
    m_format = reader.format(); 
    VERIFY_OR_DEBUG_ASSERT(!m_format.isEmpty()) {
       return {};
    }
    return ImageFileData(coverArtBytes, m_format);
```

Reply:
Oh, I see. I'm just guessing, instead of trying to read the suffix by the file name, we are reading the image and find the correct suffix from the image not name.

Another assumption, Image suffix could be changed by user and this can cause errors(?maybe) and this way we have a better approach, right?

Reply:
Yes, you mean if the user renames the png to jpg ... this is a nice side-effect :-)